Firestore/Example/Tests/Integration/API/FIRQueryTests.mm

```diff
@@ -1180,4 +1190,102 @@ - (void)testOrderByEquality {
+    deletedDocumentIds = [deletedDocumentIdsAccumulator copy];
+    XCTestExpectation *expectation = [self expectationWithDescription:@"DeleteTransaction"];
+    [collRef.firestore
+        runTransactionWithBlock:^id _Nullable(FIRTransaction *transaction, NSError **) {
+          for (NSUInteger i = 0; i < createdDocuments.count; i += 2) {
+            FIRDocumentReference *documentToDelete = createdDocuments[i];
+            [transaction deleteDocument:documentToDelete];
+            [deletedDocumentIdsAccumulator addObject:documentToDelete.documentID];
+          }
+          return @"document deletion successful";
+        }
+        completion:^(id, NSError *) {
+          [expectation fulfill];
+        }];
+    [self awaitExpectation:expectation];
+    deletedDocumentIds = [NSSet setWithArray:deletedDocumentIdsAccumulator];
+  }
+  XCTAssertEqual(deletedDocumentIds.count, 50u, @"deletedDocumentIds has the wrong size");
+
+  // Wait for 10 seconds, during which Watch will stop tracking the query and will send an existence
+  // filter rather than "delete" events when the query is resumed.
+  [NSThread sleepForTimeInterval:10.0f];
+
+  // Resume the query and save the resulting snapshot for verification.
+  FIRQuerySnapshot *querySnapshot2 = [self readDocumentSetForRef:collRef
+                                                          source:FIRFirestoreSourceDefault];
+
+  // Verify that the snapshot from the resumed query contains the expected documents; that is,
+  // that it contains the 50 documents that were _not_ deleted.
+  // TODO(b/270731363): Remove the "if" condition below once the Firestore Emulator is fixed to
+  // send an existence filter. At the time of writing, the Firestore emulator fails to send an
+  // existence filter, resulting in the client including the deleted documents in the snapshot
+  // of the resumed query.
+  if (!([FSTIntegrationTestCase isRunningAgainstEmulator] && querySnapshot2.count == 100)) {
+    NSSet<NSString *> *actualDocumentIds;
+    {
+      NSMutableArray<NSString *> *actualDocumentIdsAccumulator = [[NSMutableArray alloc] init];
+      for (FIRDocumentSnapshot *documentSnapshot in querySnapshot2.documents) {
+        [actualDocumentIdsAccumulator addObject:documentSnapshot.documentID];
+      }
+      actualDocumentIds = [NSSet setWithArray:actualDocumentIdsAccumulator];
+    }
+    NSSet<NSString *> *expectedDocumentIds;
+    {
+      NSMutableArray<NSString *> *expectedDocumentIdsAccumulator = [[NSMutableArray alloc] init];
+      for (FIRDocumentReference *documentRef in createdDocuments) {
+        if (![deletedDocumentIds containsObject:documentRef.documentID]) {
+          [expectedDocumentIdsAccumulator addObject:documentRef.documentID];
+        }
+      }
+      expectedDocumentIds = [NSSet setWithArray:expectedDocumentIdsAccumulator];
+    }
+    if (![actualDocumentIds isEqualToSet:expectedDocumentIds]) {
```

Comment:
🤔 , `actualDocumentIds` should equal to `expectedDocumentIds`, right?  why the `!`?

Reply:
should we also assert that the local cache count is 100 and existence filter count is 50?

Reply:
I mean the if condition `![actualDocumentIds isEqualToSet:expectedDocumentIds]`, this is checking if actual documents doesn't equal to expected documents, which is "sth went wrong"